bin/detect

```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+#!/usr/bin/env bash
+
+echo "-----> VulsBuildpack/detect"
```

Comment:
at some point does this become a heuristic scan of deploy targe for known types or does the buildpack plan to be forever explicitly invoked?

Reply:
I had read some recommendations that for meta-buildpack and multi-buildpack style that is opt-in only, advisable to code it that way. That being said, are you asking if we should build-in a detection pass that recursively looks in the staged buildpack files for a support dep file?

Before, I would have thought this possible. It was your link to the Brett Mogilefsky presentation at the CF2018 Summit and our follow-up discussion Tuesday that makes me think that might not work. In short: our buildpack has to exist before the others do, and stages a hook file into others later to be seen, but it does not see at the detect lifecycle phase when `bin/detect` is called any of the Python files from our 10x-dux-app for example, and especially the `Pipfile` it needs to look for.

Reply:
Also, this review comment was useful in a unrelated way: you clued me into the double shebang.

Reply:
I am not going to close this yet, but I am opening up PR for review. I think we can wait for Bret's viewpoint and decide if we can do this, and prioritize.